app/models/esdl_suite_id.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+# Represents a Users OpenId for the ESDL Suite
+class EsdlSuiteId < ApplicationRecord
+  belongs_to  :user
+
+  validates   :access_token, presence: true
+  validates   :refresh_token, presence: true
+
+  def expired?
+    return true unless expires_at
+
+    expires_at < Time.zone.now
+  end
+
+  # Refreshes the EsdlSuiteId OpenId tokens if possible, else removes this EsdlSuiteId
+  def refresh
+    new_token = EsdlSuiteService.setup.refresh(to_access_token)
+    return delete if new_token.empty?
+
+    update(new_token)
+  end
+
+  # Returns true if the EsdlSuiteId can be used for communication
+  # Returns false if the EsdlSuiteId has expired and is unable to refresh
+  def fresh?
+    refresh if expired?
```

Comment:
I actually struggled a bit with the naming here myself, so I understand where you are coming from.

> I feel as though a predicate method shouldn't have side-effects

Never thought about that, but now that you say it I agree. Such a method feels like a simple check you do, not someting that may delete the instance you are checking.

I added some more comments explaining what `refresh` and `fresh?` do, and renamed that last method to `try_viable`. Because I feel that is really what you want to know - is this `EsdlSuiteId` viable (not expired, or just refreshed), or is it already dead? But maybe there's even a better/simpler word for that in English...